### ISSUE TITLE
training speed slow

### ISSUE THREAD

**Issue by xldagger** [phantrdat/cvpr20-scatter-text-recognizer#1]
Thanks for your great job! 
could you tell me, how many hours it took you for training your model with 5 blocks?

i have tried training model on one Tesla V100 GPU with 32GB memory and without pre-training, batch_size=128, average speed is 7s/it.

here is whole training opts :
```
------------ Options -------------
exp_name: MJ-ST-SA-20-Seed2222
train_data: scatter_dataset/data_lmdb_release/training
valid_data: scatter_dataset/data_lmdb_release/validation
manualSeed: 2222
workers: 4
batch_size: 128
num_iter: 1000000
valInterval: 500
saved_model:
FT: False
adam: False
lr: 1
beta1: 0.9
rho: 0.95
eps: 1e-08
grad_clip: 5.0
select_data: ['MJ', 'ST', 'SA']
batch_ratio: ['0.4', '0.4', '0.2']
total_data_usage_ratio: 1.0
batch_max_length: 35
imgH: 32
imgW: 100
rgb: False
character: 0123456789abcdefghijklmnopqrstuvwxyz
sensitive: False
PAD: False
data_filtering_off: True
num_fiducial: 20
input_channel: 1
output_channel: 512
hidden_size: 512
log: ./saved_models/MJ-ST-SA-20-Seed2222
lambda_ctc: 0.1
lambda_attn: 1.0
num_gpu: 1
num_class: 38
```

**Comment by phantrdat** [comment]
Hi @xldagger, 
Thank you for your question, 

I remember that it had taken about 2-3 days for training this model. The training speed is also quite slow with my experience. I used 2 Tesla V100 GPUs.

**Comment by xldagger** [comment]
> Hi @xldagger,
> Thank you for your question,
> 
> I remember that it had taken about 2-3 days for training this model. The training speed is also quite slow with my experience. I used 2 Tesla V100 GPUs.

Thanks for your reply! 
so your batch_size is (64/ per GPU) * 2, and the iter nums is default 1000000?

Could you show me the opts? or could you tell me the opts which are different from default in train.py?

**Comment by phantrdat** [comment]
@xldagger The iter nums are 1000000 but I remember that I had stopped the training process earlier. Don't worry about it. Your batch size in the option is correct. I use 2 GPUs so the batch size is 128*2 = 256.

**Comment by Letdreamfly** [comment]
> Hi @xldagger,
> Thank you for your question,
> 
> I remember that it had taken about 2-3 days for training this model. The training speed is also quite slow with my experience. I used 2 Tesla V100 GPUs.

when i validate the model during training, i only select the last preds(predss[-1]) just like the inference process. And the speed is a little faster.

**Comment by Letdreamfly** [comment]
> Hi @xldagger,
> Thank you for your question,
> 
> I remember that it had taken about 2-3 days for training this model. The training speed is also quite slow with my experience. I used 2 Tesla V100 GPUs.

i only use the last preds in predss when i validate the model during train (just like the inference process), and it get a little faster speed.

**Comment by Mountchicken** [comment]
I had suffered the problem of slow training speed before. And it turns out to be the problem of shuffle in DataLoader. There are millions of pictures in MJ+ST, and random reading is slow. So PyTorch spends most of the time reading pictures and causes a low utilization rate of GPU. You can either buy an SSD or use a randomsequentialSampler used in MoranV2. Or simply just turn off the shuffle in dataloader
